test/e2e/scoped_client_test.go

```diff
@@ -0,0 +1,132 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/scoped"
+)
+
+func TestScopedClient(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, false)
+
+	require.NotEmpty(t, kubeConfigPath)
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeConfigPath)
+	require.NoError(t, err)
+
+	kubeclient := newKubeClient(t)
+	crclient := newCRClient(t)
+
+	logger := logrus.New()
+
+	tests := []struct {
+		name       string
+		grant      func(namespace, name string) (cleanup cleanupFunc)
+		assertFunc func(errGot error)
+	}{
+		// The parent test invokes 'Get' API on non existent objects. If the
+		// scoped client has enough permission, we expect a NotFound error code.
+		// Otherwise, we expect a 'Forbidden' error code due to lack of permission.
+		{
+			// The service account does not have any permission granted to it.
+			// We expect the get api call to return 'Forbidden' error due to
+			// lack of permission.
+			name: "ServiceAccountDoesNotHaveAnyPermission",
+			assertFunc: func(errGot error) {
+				require.True(t, k8serrors.IsForbidden(errGot))
+			},
+		},
+		{
+			// The service account does have permission granted to it.
+			// We expect the get api call to return 'NotFound' error.
+			name: "ServiceAccountHasPermission",
+			grant: func(namespace, name string) (cleanup cleanupFunc) {
+				cleanup = grantPermission(t, kubeclient, namespace, name)
+				return
+			},
+			assertFunc: func(errGot error) {
+				require.True(t, k8serrors.IsNotFound(errGot))
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Steps:
+			// 1. Create a new namespace
+			// 2. Create a service account.
+			// 3. Grant permission(s) to the service account if specified.
+			// 4. Get scoped client instance(s)
+			// 5. Invoke Get API call on non existent object(s) to check if
+			//    the call can be made successfully.
+			namespace := genName("a")
+			_, ceanupNS := newNamespace(t, kubeclient, namespace)
+			defer ceanupNS()
+
+			saName := genName("user-defined-")
+			sa, cleanupSA := newServiceAccount(t, kubeclient, namespace, saName)
+			defer cleanupSA()
+
+			waitServiceAccountSecretAvailable(t, kubeclient, sa.GetNamespace(), sa.GetName())
+
+			strategy := scoped.NewClientStrategy(logger, config, kubeclient, crclient)
+			getter := func() (reference *corev1.ObjectReference, err error) {
+				reference = &corev1.ObjectReference{
+					Namespace: namespace,
+					Name:      saName,
+				}
+
+				return
+			}
+
+			if tt.grant != nil {
+				cleanupPerm := tt.grant(sa.GetNamespace(), sa.GetName())
+				defer cleanupPerm()
+			}
+
+			// We expect to get scoped client instance(s).
+			kubeclientGot, crclientGot, errGot := strategy.ProvideClient(getter)
+			require.NoError(t, errGot)
```

Comment:
Do we need to test the `ProvideClient` method of the client strategy? I would think this test should focus only on the scoped client's `Get` method.